File: BikeHistory/Models/Types/EquipmentType.cs
namespace BikeHistory.Models.Types;

public class EquipmentType
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public Category Category { get; set; }
    public Guid UserId { get; set; }

    public EquipmentType(string name, Category category, Guid userId)
    {
        Name = name;
        Category = category;
        UserId = userId;
    }
}

File: frontend/src/models/equipmentTypes/equipmentType.ts
import {Category} from "@/models/equipmentTypes/category";

export interface IEquipmentType {
    id: string;
    name: string;
    category: Category;
}

export class EquipmentType implements IEquipmentType {
    id: string;
    name: string;
    category: Category;

    constructor(...args: IEquipmentType[]) {
        this.id = "";
        this.name = "";
        this.category = Category.Bike;
        if (args.length === 1) {
            Object.assign(this, args[0]);
        }
    }
}